src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -464,8 +464,23 @@ static void window_editor_object_selection_mouseup(rct_window* w, rct_widgetinde
         case WIDX_CLOSE:
             if (gScreenFlags & SCREEN_FLAGS_EDITOR)
             {
-                auto loadOrQuitAction = LoadOrQuitAction(LoadOrQuitModes::OpenSavePrompt, PM_SAVE_BEFORE_QUIT);
-                GameActions::Execute(&loadOrQuitAction);
+                window_close(w);
+                if (gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER)
+                {
+                    set_every_ride_type_invented();
```

Comment:
Is this a copy of the code that is executed when you press the "Next" button? If so, it should be a function that is called from both locations.

Reply:
Done. I've also changed it so the x doesn't open up a map or the ride selection window like the next button.